packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,410 @@
+  const [stylesheet, setStylesheet] = useState<string | null>(null);
+  const [validator, setValidator] = useState<string>('AJV8');
+  const [showForm, setShowForm] = useState(false);
+  const [liveSettings, setLiveSettings] = useState<LiveSettings>({
+    showErrorList: 'top',
+    validate: false,
+    disable: false,
+    readonly: false,
+    omitExtraData: false,
+    liveOmit: false,
+  });
+  const [FormComponent, setFormComponent] = useState<ComponentType<FormProps>>(withTheme({}));
+  const [ArrayFieldTemplate, setArrayFieldTemplate] = useState<ComponentType<ArrayFieldTemplateProps>>();
+  const [ObjectFieldTemplate, setObjectFieldTemplate] = useState<ComponentType<ObjectFieldTemplateProps>>();
+
+  const playGroundFormRef = useRef<any>(null);
+
+  const onThemeSelected = useCallback(
+    (theme: string, { stylesheet, theme: themeObj }: { stylesheet?: any; theme?: any }) => {
+      setTheme(theme);
+      setSubtheme(null);
+      setFormComponent(withTheme(themeObj));
+      setStylesheet(stylesheet);
+    },
+    [setTheme, setSubtheme, setFormComponent, setStylesheet]
+  );
+
+  const load = useCallback(
+    (data: any) => {
+      // Reset the ArrayFieldTemplate whenever you load new data
+      const { ArrayFieldTemplate, ObjectFieldTemplate, extraErrors } = data;
+      // uiSchema is missing on some examples. Provide a default to
+      // clear the field in all cases.
+      const { uiSchema = {} } = data;
+
+      const { theme: dataTheme = theme } = data;
+
+      onThemeSelected(dataTheme, themes[dataTheme]);
+
+      // force resetting form component instance
+      setShowForm(false);
+
+      setUiSchema(uiSchema);
+      setExtraErrors(extraErrors);
+      setTheme(dataTheme);
+      setArrayFieldTemplate(ArrayFieldTemplate);
+      setObjectFieldTemplate(ObjectFieldTemplate);
+      setShowForm(true);
+    },
+    [onThemeSelected, setShowForm, setUiSchema, setExtraErrors, setTheme, setArrayFieldTemplate, setObjectFieldTemplate]
+  );
+
+  useEffect(() => {
+    const hash = document.location.hash.match(/#(.*)/);
+
+    if (hash && typeof hash[1] === 'string' && hash[1].length > 0) {
+      try {
+        load(JSON.parse(atob(hash[1])));
+      } catch (error) {
+        alert('Unable to load form setup data.');
+        console.error(error);
+      }
+
+      return;
+    }
+
+    // initialize theme
+    onThemeSelected(theme, themes[theme]);
+
+    setShowForm(true);
+  }, [onThemeSelected, load, setShowForm]);
+
+  const onSubthemeSelected = useCallback(
+    (subtheme: any, { stylesheet }: { stylesheet: any }) => {
+      setSubtheme(subtheme);
+      setStylesheet(stylesheet);
+    },
+    [setSubtheme, setStylesheet]
+  );
+
+  const onValidatorSelected = useCallback(
+    (validator: string) => {
+      setValidator(validator);
+    },
+    [setValidator]
+  );
+
+  const handleSetLiveSettings = useCallback(
+    ({ formData }: { formData: any }) => {
```

Comment:
This should alleviate the cast below
```suggestion
    ({ formData }: IChangeEvent<any, RJSFSchema, any>) => {
```